File: Atm.java


import java.util.Locale;
import java.util.Scanner;

public class ATMSimulator {
    private static long bacaAngka(Scanner input, String pesan) {
        while (true) {
            System.out.print(pesan);
            if (input.hasNextLong()) {
                long nilai = input.nextLong();
                input.nextLine(); // clear line
                return nilai;
            }
            System.out.println("Input tidak valid. Masukkan angka saja.");
            input.nextLine(); 
        }
    }

    private static String rupiah(long jumlah) {
        return String.format(Locale.forLanguageTag("id-ID"), "Rp %,d", jumlah);
    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        long saldo = 1_000_000; // (Saldo awal)
        long pilihan;

        System.out.println("=== Selamat datang di ATM ===");

        // Loop:
        do {
            // ATM menu
            System.out.println();
            System.out.println("------ MENU ATM ------");
            System.out.println("1. Cek Saldo");
            System.out.println("2. Setor Uang");
            System.out.println("3. Tarik Uang");
            System.out.println("4. Keluar");
            System.out.println("----------------------");

            // Pilih menu
            pilihan = bacaAngka(input, "Pilih menu (1-4): ");

            switch ((int) pilihan) {
                case 1:
                    // tampil saldo
                    System.out.println("Saldo Anda saat ini: " + rupiah(saldo));
                    break;

                case 2:
                    // Deposit
                    long setor = bacaAngka(input, "Jumlah setor: Rp ");
                    if (setor > 0) {
                        saldo = saldo + setor;
                        System.out.println("Setor berhasil. Saldo sekarang: " + rupiah(saldo));
                    } else {
                        System.out.println("Jumlah setor harus lebih dari 0.");
                    }
                    break;

                case 3:
                    // tarik uang kalo cukup
                    long tarik = bacaAngka(input, "Jumlah tarik: Rp ");
                    if (tarik <= 0) {
                        System.out.println("Jumlah tarik harus lebih dari 0.");
                    } else if (tarik > saldo) {
                        System.out.println("Saldo tidak mencukupi. Saldo Anda: " + rupiah(saldo));
                    } else {
                        saldo = saldo - tarik;
                        System.out.println("Tarik berhasil. Saldo sekarang: " + rupiah(saldo));
                    }
                    break;

                case 4:
                    System.out.println("Terima kasih telah menggunakan ATM.");
                    break;

                default:
                    System.out.println("Menu tidak tersedia. Pilih angka 1 sampai 4.");
            }
        } while (pilihan != 4); // balik menu awal

        // tunjukin saldo akhir
        System.out.println("Saldo akhir Anda: " + rupiah(saldo));
        input.close();
    }
}

File: atm.js

let saldo = 0; 
let pilihan;          
let jumlah;            
let inputPilihan;      
let inputJumlah;       

console.log("=== SELAMAT DATANG DI ATM ===");

do {
  console.log("");
  console.log("------ MENU ATM ------");
  console.log("1. Cek Saldo");
  console.log("2. Setor Uang");
  console.log("3. Tarik Uang");
  console.log("4. Keluar");
  console.log("----------------------");

  inputPilihan = prompt("Pilih menu (1-4): ");
  pilihan = Number(inputPilihan);

  if (isNaN(pilihan)) {
    console.log("Input tidak valid. Masukkan angka saja.");
  } else {
    switch (pilihan) {
      case 1:
        // Cek saldo
        console.log("Saldo Anda saat ini: Rp " + saldo.toLocaleString("id-ID"));
        break;

      case 2:
        // Setor uang
        inputJumlah = prompt("Jumlah setor: Rp ");
        jumlah = Number(inputJumlah);
        if (isNaN(jumlah)) {
          console.log("Input tidak valid. Masukkan angka saja.");
        } else if (jumlah > 0) {
          saldo = saldo + jumlah;
          console.log("Setor berhasil. Saldo sekarang: Rp " + saldo.toLocaleString("id-ID"));
        } else {
          console.log("Jumlah setor harus lebih dari 0.");
        }
        break;

      case 3:
        // Tarik uang
        inputJumlah = prompt("Jumlah tarik: Rp ");
        jumlah = Number(inputJumlah);
        if (isNaN(jumlah)) {
          console.log("Input tidak valid. Masukkan angka saja.");
        } else if (jumlah <= 0) {
          console.log("Jumlah tarik harus lebih dari 0.");
        } else if (jumlah > saldo) {
          console.log("Saldo tidak mencukupi. Saldo Anda: Rp " + saldo.toLocaleString("id-ID"));
        } else {
          saldo = saldo - jumlah;
          console.log("Tarik berhasil. Saldo sekarang: Rp " + saldo.toLocaleString("id-ID"));
        }
        break;

      case 4:
        console.log("Terima kasih telah menggunakan ATM.");
        break;

      default:
        console.log("Menu tidak tersedia. Pilih angka 1 sampai 4.");
    }
  }
} while (pilihan !== 4);

console.log("");
console.log("Saldo akhir Anda: Rp " + saldo.toLocaleString("id-ID"));